index.js

```diff
@@ -0,0 +1,318 @@
+'use strict';
+
+const T = require('sanctuary-def');
+
+const AutoPredicate = f =>
+  T.NullaryType(`[${f.toString()}]`, f);
+
+const B = (f, g) => (...args) => f(g(...args));
+
+const map = require('ramda/src/map');
+const curryN = require('ramda/src/curryN');
+
+const values = o =>
+  Array.isArray(o)
+    ? o
+    : Object.keys(o).map(k => o[k]);
+
+const zipObj = ks => vs =>
+  ks.length
+    ? Object.assign(
+      ...ks.map((k, i) => ({[k]: vs[i]}))
+    )
+    : {};
+
+const unapply = f => (...values) => f(values);
+
+const mapConstrToFn = constraint =>
+
+  constraint === String
+    ? T.String
+  : constraint === Number
+    ? T.Number
+  : constraint === Boolean
+    ? T.Boolean
+  : constraint === Object
+    ? T.Object
+  : constraint === Array
+    ? T.Array(T.Any)
+  : constraint === Function
+    ? T.AnyFunction
+    : constraint;
+
+const BuiltInType = function(t){
+  const mapped = mapConstrToFn(t);
+
+  if (mapped !== t){
+    return mapped;
+  } else if (t.constructor === Function) {
```

Comment:
Let's not rely on object identity for the reasons given in sanctuary-js/sanctuary#100. `typeof` should be fine here.